Zend/Zend.m4

```diff
@@ -182,16 +182,33 @@ AS_VAR_IF([ZEND_DEBUG], [yes], [
   fi
 ], [AC_DEFINE([ZEND_DEBUG], [0])])
 
-test -n "$GCC" && CFLAGS="-Wall -Wextra -Wno-unused-parameter -Wno-sign-compare $CFLAGS"
-dnl Check if compiler supports -Wno-clobbered (only GCC)
-AX_CHECK_COMPILE_FLAG([-Wno-clobbered], CFLAGS="-Wno-clobbered $CFLAGS", , [-Werror])
-dnl Check for support for implicit fallthrough level 1, also add after previous CFLAGS as level 3 is enabled in -Wextra
-AX_CHECK_COMPILE_FLAG([-Wimplicit-fallthrough=1], CFLAGS="$CFLAGS -Wimplicit-fallthrough=1", , [-Werror])
-AX_CHECK_COMPILE_FLAG([-Wduplicated-cond], CFLAGS="-Wduplicated-cond $CFLAGS", , [-Werror])
-AX_CHECK_COMPILE_FLAG([-Wlogical-op], CFLAGS="-Wlogical-op $CFLAGS", , [-Werror])
-AX_CHECK_COMPILE_FLAG([-Wformat-truncation], CFLAGS="-Wformat-truncation $CFLAGS", , [-Werror])
-AX_CHECK_COMPILE_FLAG([-Wstrict-prototypes], CFLAGS="-Wstrict-prototypes $CFLAGS", , [-Werror])
-AX_CHECK_COMPILE_FLAG([-fno-common], CFLAGS="-fno-common $CFLAGS", , [-Werror])
+AS_VAR_IF([GCC], [yes],
+  [CFLAGS="-Wall -Wextra -Wno-unused-parameter -Wno-sign-compare $CFLAGS"])
+
+dnl Check if compiler supports -Wno-clobbered (only GCC).
+AX_CHECK_COMPILE_FLAG([-Wno-clobbered],
+  [CFLAGS="-Wno-clobbered $CFLAGS"],,
+  [-Werror])
+dnl Check for support for implicit fallthrough level 1, also add after previous
+dnl CFLAGS as level 3 is enabled in -Wextra.
+AX_CHECK_COMPILE_FLAG([-Wimplicit-fallthrough=1],
+  [CFLAGS="$CFLAGS -Wimplicit-fallthrough=1"],,
+  [-Werror])
```

Comment:
Complete aside, I need to look at what the differences in levels are an if we maybe should increase it.